packages/clerk-js/src/core/clerk.ts

```diff
@@ -1504,7 +1505,8 @@ export class Clerk implements ClerkInterface {
   };
 
   #loadInStandardBrowser = async (): Promise<boolean> => {
-    this.#authService = new AuthCookieService(this, this.#fapiClient);
+    const cookieSuffix = await getCookieSuffix(this.publishableKey);
+    this.#authService = new AuthCookieService(this, this.#fapiClient, cookieSuffix);
```

Comment:
Do you think we should deal with cookie suffixes inside `AuthCookieService`?
Does `Clerk` need to know about this? We don't seem to use it anywhere else here. `AuthCookieService` does hold a reference to `this`, so we can read the publishableKey from within. 

If you want to leverage DI for testing, maybe pass the `getCookieSuffix` function instead?

Reply:
The reason i am passing the cookieSuffix value in `AuthCookieService` is because we cannot initiliaze the cookieSuffix value in the AuthCookieService contrustor where the cookie handlers are initialized and use the cookie suffix since the generation of the cookie suffix is async due to the crypto sha1 addition.

Reply:
You're right! The alternative here would be to defer the init logic into an `init` method instead, something like 

```
this.#authService = new AuthCookieService(this, this.#fapiClient);
await this.#authService.init()
```

Would that be better for our case? Its a minor difference, but the singleton would have one less dependency to worry about.

Reply:
I am not against that approach. The risky part in this implementation is that we should be very careful to call the `init()` method exactly after the service initialization or enforce somehow (eg throwing an error) that the `init()` method is called before the authService is used.
If we decide to use the approach of introducing a method to initialize the authService instead of depending in the constructor, i would suggest we expose a static method to initialize the instance and make the constructor private.
cc: @nikosdouvlis

Reply:
Agreed, I think its better in this case. We can always revise this decision later

Reply:
Implementation updated to use private constructor and static method to initialize the AuthCookieService.
Reasons behind updating implementation as suggested by Nikos:
- reduce dependencies in core/clerk.ts
- AuthCookieService contains all logic for cookie and cookie suffixes
- a way to bypass the limitation of un-supported async constructors